src/ItemsPerHoldingsRecord.js

```diff
@@ -55,59 +54,121 @@ class ItemsPerHoldingsRecord extends React.Component {
-    if (e) e.stopPropagation();
-    this.props.mutator.addItemMode.replace({ mode: true });
-    this.addItemModeThisLayer = true;
+    const { mutator } = this.props;
+
+    if (e) {
+      e.preventDefault();
+      e.stopPropagation();
+    }
+
+    mutator.addItemMode.replace({ mode: true });
+    mutator.query.update({ layer: 'createItem' });
   }
 
   onClickCloseNewItem = (e) => {
+    const { mutator } = this.props;
+
     if (e) e.preventDefault();
-    this.props.mutator.addItemMode.replace({ mode: false });
-    this.addItemModeThisLayer = false;
+
+    mutator.addItemMode.replace({ mode: false });
+    mutator.query.update({ layer: null });
   }
 
   createItem = (item) => {
-    // POST item record
     this.props.mutator.items.POST(item);
     this.onClickCloseNewItem();
   }
 
+  renderButtonsGroup = () => {
+    const {
+      instance,
+      holdingsRecord,
+      stripes: {
+        intl: {
+          formatMessage,
+        }
+      }
+    } = this.props;
+
+    return (
+      <div>
+        <Button
+          to={{ pathname: `/inventory/view/${instance.id}/${holdingsRecord.id}` }}
+          style={{ marginRight: '5px' }}
+        >
+          { formatMessage({ id: 'ui-inventory.viewHoldings' }) }
+        </Button>
+        <Button
+          onClick={this.onClickAddNewItem}
+          buttonStyle="primary paneHeaderNewButton"
+        >
+          { formatMessage({ id: 'ui-inventory.addItem' }) }
+        </Button>
+      </div>
+    );
+  }
+
   render() {
-    const { okapi, referenceTables, resources: { addItemMode, materialTypes, loanTypes }, instance, holdingsRecord, accordionToggle, accordionStates } = this.props;
+    const {
+      okapi,
+      referenceTables,
+      resources: { materialTypes, loanTypes, query },
+      instance,
+      holdingsRecord,
+      accordionToggle,
+      accordionStates,
+    } = this.props;
     const { locationsById } = referenceTables;
     const materialtypes = (materialTypes || {}).records || [];
     const loantypes = (loanTypes || {}).records || [];
 
     referenceTables.loanTypes = loantypes;
     referenceTables.materialTypes = materialtypes;
 
-    const viewHoldingsPath = `/inventory/view/${this.props.instance.id}/${this.props.holdingsRecord.id}`;
-
     const formatMsg = this.props.stripes.intl.formatMessage;
-    const viewHoldingsButton = <Link to={viewHoldingsPath}><Button id="clickable-view-holdings">{formatMsg({ id: 'ui-inventory.viewHoldings' })}</Button></Link>;
-    const newItemButton = <Button id="clickable-new-item" onClick={this.onClickAddNewItem} title={formatMsg({ id: 'ui-inventory.addItem' })} buttonStyle="primary paneHeaderNewButton">{formatMsg({ id: 'ui-inventory.addItem' })}</Button>;
     const labelLocation = holdingsRecord.permanentLocationId ? locationsById[holdingsRecord.permanentLocationId].name : '';
     const labelCallNumber = holdingsRecord.callNumber || '';
 
+    if (query.layer === 'createItem') {
+      return (
+        <Layer
```

Comment:
I'm wondering - if these will be rendered at separately routed URLs - do we still need to wrap it in a `<Layer>` component?

Reply:
Yes, `Layer` is needed to render the component over the main page and stretched to full width of the browser window.